targets/nrf5x/bluetooth.c

```diff
@@ -2524,16 +2531,18 @@ uint32_t jsble_advertising_start() {
 
   //jsiConsolePrintf("adv_data_set %d %d\n", advPtr, advLen);
 #if NRF_SD_BLE_API_VERSION>5
-  ble_gap_adv_data_t d;
-  memset(&d, 0, sizeof(d));
-  d.adv_data.p_data = (uint8_t*)advPtr;
-  d.adv_data.len = advLen;
+  advLen=MIN(advLen,BLE_GAP_ADV_SET_DATA_SIZE_MAX);
+  memset(&m_ble_gap_adv_data, 0, sizeof(m_ble_gap_adv_data));
+  memcpy(m_adv_data, advPtr, advLen);
+  m_ble_gap_adv_data.adv_data.p_data = m_adv_data;
+  m_ble_gap_adv_data.adv_data.len = advLen;
   if (!non_scannable) {
-    d.scan_rsp_data.p_data = m_enc_scan_response_data;
-    d.scan_rsp_data.len = m_enc_scan_response_data_len;
+    memcpy(m_scan_rsp_data, m_enc_scan_response_data, m_enc_scan_response_data_len);
+    m_ble_gap_adv_data.scan_rsp_data.p_data = m_scan_rsp_data;
+    m_ble_gap_adv_data.scan_rsp_data.len = m_enc_scan_response_data_len;
   }
 
-  err_code = sd_ble_gap_adv_set_configure(&m_adv_handle, &d, &adv_params);
+  err_code = sd_ble_gap_adv_set_configure(&m_adv_handle, &m_ble_gap_adv_data, &adv_params);
```

Comment:
I know this is the first use so `jsble_advertising_update` wouldn't work as-is, but it'd be nice if we could find a way to remove the duplication here (and in the ifdefs below where we do sd_ble_gap_adv_data_set as well).

I guess if we `m_ble_gap_adv_data` was all zero'd out it'd be good enough.

Reply:
m_ble_gap_adv_data  is in bss so is zeroed out on startup but then it is not after stopping advertising or any time later, this one does not need to be static for SD>6.1 to work but it keeps enough data for making the temporary copy and when you want to set just the other half

Reply:
Leaving the structure is fine by me - also the JS vars do get tricky when we're trying to configure stuff as we tear down and restart the JS engine :)

What I meant was just whether we could use `jsble_advertising_update` here again to reduce some duplication since they are almost identical.

Reply:
oh, got it, but don't know how, it does set it twice with temporary copy because the guess is that it is already advertising at that point but that may not be true. when properly knowing it is not advertising it could skip that and then it could be used

Reply:
could probably use ` bleStatus & BLE_IS_ADVERTISING`, will test

Reply:
Or I was thinking if both the lengths of data are zero (eg m_ble_gap_adv_data was memset(0) or cleared by reboot) then it could assume there was no need for that first call.

Reply:
Sounds good - thanks!